tests/ga/test_extension.py

```diff
@@ -2535,304 +2536,285 @@ def test_ext_handler_with_systemd_cgroup_enabled(self, *args):
+        first_ext.get_command("uninstallCommand").assert_called_once()
+        second_ext.get_command("installCommand").assert_called_once()
+        
+        _, kwargs = second_ext.get_command("updateCommand").call_args
+
+        assert kwargs.get("env", {}).get(ExtCommandEnvVariable.DisableReturnCode, "") == str(exit_code)
+
+
+    def test_uninstall_failed_env_variable_should_set_for_install_when_continue_on_update_failure_is_true(self):
+
+        exit_code = uuid.uuid4()    # Generate a unique UUID for highest strictness.
+        unique_fail = lambda *args, **kwargs: exit_code
+
+        first_ext = get_extension_actor(uninstallAction=unique_fail)
+        second_ext = get_extension_actor(version="1.1.0", continueOnUpdateFailure=True)
+
+        TestExtensionUpdateOnFailure._do_upgrade_scenario(first_ext, second_ext)
+
+        first_ext.get_command("disableCommand").assert_called_once()            
+        second_ext.get_command("updateCommand").assert_called_once()
+        first_ext.get_command("uninstallCommand").assert_called_once()
+        second_ext.get_command("installCommand").assert_called_once()
+        
+        _, kwargs = second_ext.get_command("installCommand").call_args
+
+        assert kwargs.get("env", {}).get(ExtCommandEnvVariable.UninstallReturnCode, "") == str(exit_code)
 
-            self.assertTrue('-install' in kwargs['command'] and ExtCommandEnvVariable.UninstallReturnCode in kwargs['env'],
-                            "The install command should have Uninstall Failed in env variable")
 
     def test_extension_error_should_be_raised_when_continue_on_update_failure_is_false_on_disable_failure(self, *args):
-        old_handler_i = self._get_ext_handler_instance('foo', '1.0.0')
-        new_handler_i = self._get_ext_handler_instance('foo', '1.0.1', continue_on_update_failure=False)
+        exit_code = uuid.uuid4()    # Generate a unique UUID for highest strictness.
+        unique_fail = lambda *args, **kwargs: exit_code
 
-        with patch.object(ExtHandlerInstance, "disable", side_effect=ExtensionError("Disable Failed")):
-            with self.assertRaises(ExtensionUpdateError) as error:
-                # Ensure the error is of type ExtensionUpdateError
-                ExtHandlersHandler._update_extension_handler_and_return_if_failed(old_handler_i, new_handler_i)
+        first_ext = get_extension_actor(disableAction=unique_fail)
+        second_ext = get_extension_actor(version="1.1.0", continueOnUpdateFailure=False)
+
+        TestExtensionUpdateOnFailure._do_upgrade_scenario(first_ext, second_ext)
+            
+        first_ext.get_command("disableCommand").assert_called_once()
+        second_ext.get_command("updateCommand").assert_not_called()
+        first_ext.get_command("uninstallCommand").assert_not_called()
+        second_ext.get_command("installCommand").assert_not_called()
+
+        assert len(second_ext.statusBlobs) == 1
```

Comment:
Please use self.assertEquals, self.assertTrue, etc instead of this syntax. The files we own use self.*, so let's be consistent.